war/src/main/java/io/lumeer/engine/controller/search/QuerySuggester.java

```diff
@@ -0,0 +1,186 @@
+ */
+@ApplicationScoped
+public class QuerySuggester {
+
+   private static final int SUGGESTIONS_LIMIT = 10;
+
+   @Inject
+   @UserDataStorage
+   private DataStorage userDataStorage;
+
+   @Inject
+   private DataStorageDialect storageDialect;
+
+   @Inject
+   private CollectionMetadataFacade collectionMetadataFacade;
+
+   @Inject
+   private ProjectFacade projectFacade;
+
+   @Inject
+   private ViewFacade viewFacade;
+
+   /**
+    * Suggests how to complete the part of the query the user is typing
+    *
+    * @param text
+    *       part of the query the user is typing
+    * @param type
+    *       suggestion type from the request
+    * @return a list of suggestions (collections, links, views, etc.)
+    */
+   public DataDocument suggest(String text, SuggestionsType type) {
+      switch (type) {
+         case ATTRIBUTE:
+            return new DataDocument(SuggestionsDocument.ATTRIBUTES, suggestAttributes(text));
+         case COLLECTION:
+            return new DataDocument(SuggestionsDocument.COLLECTIONS, suggestCollections(text));
+         case LINK:
+            return new DataDocument(SuggestionsDocument.LINKS, suggestLinkTypes(text));
+         case VIEW:
+            return new DataDocument(SuggestionsDocument.VIEWS, suggestViews(text));
+         default:
+            return suggestAll(text);
+      }
+   }
+
+   DataDocument suggestAll(String text) {
+      return new DataDocument()
+            .append(SuggestionsDocument.ATTRIBUTES, suggestAttributes(text))
+            .append(SuggestionsDocument.COLLECTIONS, suggestCollections(text))
+            .append(SuggestionsDocument.LINKS, suggestLinkTypes(text))
+            .append(SuggestionsDocument.VIEWS, suggestViews(text));
+   }
+
+   List<DataDocument> suggestCollections(String text) {
+      String metadataCollection = collectionMetadataFacade.metadataCollection();
+      DataFilter filter = storageDialect.fieldValueWildcardFilter(LumeerConst.Collection.REAL_NAME_KEY, text);
+      List<String> projection = Arrays.asList(LumeerConst.Collection.REAL_NAME_KEY);
+
+      return userDataStorage.search(metadataCollection, filter, projection, SUGGESTIONS_LIMIT)
+                            .stream()
+                            .map(QuerySuggester::convertCollection)
+                            .collect(Collectors.toList());
+   }
+
+   private static DataDocument convertCollection(DataDocument collection) {
+      return new DataDocument()
+            .append(SuggestionsDocument.COLLECTIONS_NAME, collection.getString(LumeerConst.Collection.REAL_NAME_KEY))
+            .append(SuggestionsDocument.COLLECTIONS_ICON, ""); // TODO store paths to icons in the database or somewhere else
+   }
+
+   List<DataDocument> suggestAttributes(String text) {
+      String[] parts = text.split("\\.", 2);
+      if (parts.length < 2) {
+         return suggestCollectionsForAttribute(text);
+      }
+
+      String collectionName = parts[0];
+      String attributePart = parts[1];
+
```

Comment:
I thought it would be an overkill to search for all attributes of all collections but as I am thinking of it know there will probably not be so many of them so it might not be that inefficient.